src/Command/ErrorFormatter/FailWithoutResultCacheErrorFormatter.php

```diff
@@ -0,0 +1,26 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Command\ErrorFormatter;
+
+use PHPStan\Command\AnalysisResult;
+use PHPStan\Command\Output;
+
+class FailWithoutResultCacheErrorFormatter implements ErrorFormatter
+{
+
+	public function __construct(
+		private TableErrorFormatter $tableErrorFormatter,
+	)
+	{
+	}
+
+	public function formatErrors(AnalysisResult $analysisResult, Output $output): int
+	{
+		if (!$analysisResult->isResultCacheUsed()) {
+			return 2;
```

Comment:
to differentiate a error caused by non-existant result cache, I figured I would use a separate code from 0/1 which the other formatters use